docs/products/kafka/kafka-connect/howto/s3-source-connector.md

```diff
@@ -0,0 +1,193 @@
+
+:::note
+Ensure that the `{{topic}}` placeholder does not accidentally capture the wrong part of
+the filename. For example, if `file.name.template` is `{{topic}}-log-{{partition}}.json`
+and the filename is `billing-log-1.json`, the extracted topic will
+be `billing`, not `billing-log`.
+To prevent this, use a clear delimiter, such as `{{topic}}/{{partition}}.json`.
+:::
+
+## Supported S3 object formats
+
+The connector supports four S3 object formats. Choose the one that best fits your data
+and processing needs:
+
+| Format | Description | Configuration | Example |
+|--------|------------|--------------|---------|
+| JSON Lines (`jsonl`) | Each line is a valid JSON object, commonly used for event streaming. | `input.format=jsonl` | ```json { "key": "k1", "value": "v0", "offset": 1232155, "timestamp": "2020-01-01T00:00:01Z" } ``` |
+| Avro (`avro`) | A compact, schema-based binary format for efficient serialization. | `input.format=avro` | ```json { "type": "record", "fields": [ { "name": "key", "type": "string" }, { "name": "value", "type": "string" }, { "name": "timestamp", "type": "long" } ] } ``` |
+| Parquet (`parquet`) | A columnar format optimized for fast queries. Stores data in a compressed, column-based structure.| `input.format=parquet` | Uses a schema similar to Avro but optimized for analytics.|
+| Bytes (`bytes`) (default) | A raw byte stream format for unstructured data. | `input.format=bytes` | No predefined structure |
+
+## Create an Amazon S3 source connector configuration file
+
+Create a file named `s3_source_connector.json` with the following configuration:
+
+```json
+  {
+    "name": "aiven-s3-source-connector",
+    "connector.class": "io.aiven.kafka.connect.s3.source.S3SourceConnector",
+    "aws.access.key.id": "YOUR_AWS_ACCESS_KEY_ID",
+    "aws.secret.access.key": "YOUR_AWS_SECRET_ACCESS_KEY",
+    "aws.s3.bucket.name": "your-s3-bucket-name",
+    "aws.s3.region": "your-s3-region",
+    "aws.s3.prefix": "optional/prefix/",
+    "aws.credentials.provider": "software.amazon.awssdk.auth.credentials.AwsCredentialsProviderChain",
+    "topics": "your-target-kafka-topic",
+    "file.name.template": "{{topic}}-{{partition}}-{{start_offset}}",
+    "tasks.max": 1,
+    "poll.interval.ms": 10000,
+    "error.tolerance": "all",
+    "input.format": "bytes"
+  }
+```
+
+Parameters:
+
+- `name`: A unique name for the connector.
+- `connector.class`: The Java class for the connector.
+- `aws.access.key.id` and `aws.secret.access.key`: AWS IAM credentials for
+  authentication.
+- `aws.s3.bucket.name`: The name of the S3 bucket containing the source data.
+- `aws.s3.region`: The AWS region where the bucket is located.
+- `aws.s3.prefix`: Optional. Filters objects within the S3 bucket.
+- `aws.credentials.provider`: Specifies the AWS credentials provider.
+- `topics`: The Kafka topic where the ingested data is published.
```

Comment:
```suggestion
- `topic`: The Kafka topic where the ingested data is published.
```

We can also mention that this is an optional config. If this is not configured, then topic is retrieved from file.name.template config